server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -598,16 +637,20 @@ public static void acquireFSLockForPaths(IndexSettings indexSettings, Path... sh
      * allow the folder to be deleted
      *
      * @param lock the shards lock
+     * @param shardPath the shard path of the shard to delete
      * @throws IOException if an IOException occurs
      * @throws ElasticsearchException if the write.lock is not acquirable
      */
-    public void deleteShardDirectoryUnderLock(ShardLock lock, IndexSettings indexSettings) throws IOException {
+    public void deleteShardDirectoryUnderLock(ShardLock lock, ShardPath shardPath, IndexSettings indexSettings) throws IOException {
```

Comment:
`@Nullable ShardPath shardPath`

Reply:
I reverted the modification of this method.